app/api/events.py

```diff
@@ -107,7 +108,13 @@ class Meta:
 
 
 class EventList(ResourceList):
-    decorators = (jwt_required, )
+    def query(self, view_kwargs):
+        query_ = self.session.query(Track)
+        if view_kwargs.get('user_id') is not None:
+            query_ = query_.join(User).filter(User.id == view_kwargs['user_id'])
+        return query_
+
+    decorators = (jwt_required, accessible_events, )
```

Comment:
Yes it is meant only for GET request but Thanks :D Your comment made me realise this will not work even for GET requests. Made some changes. I'm testing it and then will push it here. It is working fine as expected.